pageserver/src/tenant_mgr.rs

```diff
@@ -407,25 +388,35 @@ pub fn update_tenant_config(
     tenant_id: TenantId,
 ) -> anyhow::Result<()> {
     info!("configuring tenant {tenant_id}");
-    get_tenant(tenant_id, true)?.update_tenant_config(tenant_conf);
+    get_active_tenant(tenant_id)?.update_tenant_config(tenant_conf);
     Tenant::persist_tenant_config(&conf.tenant_config_path(tenant_id), tenant_conf, false)?;
     Ok(())
 }
 
-/// Gets the tenant from the in-memory data, erroring if it's absent or is not fitting to the query.
-/// `active_only = true` allows to query only tenants that are ready for operations, erroring on other kinds of tenants.
-pub fn get_tenant(tenant_id: TenantId, active_only: bool) -> anyhow::Result<Arc<Tenant>> {
+/// Gets the tenant in active state from the in-memory data, erroring if it's absent.
+pub fn get_active_tenant(tenant_id: TenantId) -> anyhow::Result<ActiveTenantGuard> {
     let m = tenants_state::read_tenants();
-    let tenant = m
+    let guard = m
         .get(&tenant_id)
         .with_context(|| format!("Tenant {tenant_id} not found in the local state"))?;
-    if active_only && !tenant.is_active() {
-        anyhow::bail!("Tenant {tenant_id} is not active")
-    } else {
-        Ok(Arc::clone(tenant))
+    match guard.upgrade() {
+        Some(tenant) if tenant.is_active() => Ok(tenant),
+        _ => bail!("Tenant {tenant_id} is not active"),
```

Comment:
NIT: could separate alive/active error messages here too, as in `with_context` below.